### ISSUE TITLE
Questions about the measurement of zero-shot learning uncertianty

### ISSUE THREAD

**Issue by HelloWorldLTY** [mkirchhof/url#2]
Hi, thanks for your great work. I have two questions after reading your codes and manuscript.

1. I think it measures the zero-shot learning ability, so why we still need to train the model in the train.py code? Moreover, some measurement like mcinfonce has trainable parameters, while other methods like mc dropout does not have it. Does this option bring bias to this experiment?

2. Do you think it is possible to extend the conclusion to the regression task? For example, applying mcinfonce for the representation of regression tasks?

I am looking forward to your reply!

**Comment by mkirchhof** [comment]
Hi, thanks for the kind words!

1a. We need to first pretrain the uncertainty module, since it's randomly initiated. This pretraining happens on ImageNet-21k. The evaluation happens on zero-shot datasets like CUB, CARS, ... :)

1b. In the URL paper, we tuned all those hyperparameters on a validation split (a seperate set of classes on the zero-shot datasets). In our newest paper with loss prediction, we don't even do that anymore, since it comes without hyperparameters. In any case, I do not think it biases the experiment too much: In the URL paper, we've also once optimized the methods w.r.t. not uncertainties but recall@1, and results were quite similar for most methods.

2. I think it should be possible. For MCInfoNCE most likely (will probably work if the standard InfoNCE works for regression, you might have to check that in literature), for loss prediction definitely. Loss prediction originally came from regression literature, and the framework is flexible to predict any loss, whether classification, regression, or anything else :)

Let me know if you have more questions!

**Comment by HelloWorldLTY** [comment]
I see, thanks for your explaniation. My project is not for images so I may need to pretrain a new model if I intend to use your pipeline.

**Comment by mkirchhof** [comment]
That sounds exciting! Definitely let me know what comes out, positive or negative :) I'd close this issue for the moment, but feel free to open another one if you have questions!
